Sources/Nimble/Adapters/ObjectiveC/XCTestObservationCenter+Register.m

```diff
@@ -0,0 +1,74 @@
+#import "CurrentTestCaseTracker.h"
+#import <XCTest/XCTest.h>
+#import <objc/runtime.h>
+
+#pragma mark - Private
+
+@interface XCTestObservationCenter (Private)
+- (void)_addLegacyTestObserver:(id)observer;
+@end
+
+@implementation XCTestObservationCenter (Register)
+
+/// Uses objc method swizzling to register `CurrentTestCaseTracker` as a test observer. This is necessary
+/// because Xcode 7.3 introduced timing issues where if a custom `XCTestObservation` is registered too early
+/// it suppresses all console output (generated by `XCTestLog`), breaking any tools that depend on this output.
+/// This approach waits to register our custom test observer until XCTest adds its first "legacy" observer,
+/// falling back to registering after the first normal observer if this private method ever changes.
++ (void)load {
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
```

Comment:
Dispatching isn't really needed here since the runtime provides the single-execution guarantee for the load method already.